pkg/reconciler/certificate/certificate.go

```diff
@@ -173,18 +173,30 @@ func (c *Reconciler) reconcileCMCertificate(ctx context.Context, knCert *v1alpha
 	return cmCert, nil
 }
 
-func (c *Reconciler) updateStatus(desired *v1alpha1.Certificate) (*v1alpha1.Certificate, error) {
-	cert, err := c.knCertificateLister.Certificates(desired.Namespace).Get(desired.Name)
-	if err != nil {
-		return nil, err
-	}
-	// If there's nothing to update, just return.
-	if reflect.DeepEqual(cert.Status, desired.Status) {
-		return cert, nil
-	}
-	// Don't modify the informers copy
-	existing := cert.DeepCopy()
-	existing.Status = desired.Status
+func (c *Reconciler) updateStatus(desired *v1alpha1.Certificate) error {
+	return reconciler.RetryUpdateConflicts(func(attempts int) error {
+		var (
+			existing *v1alpha1.Certificate
+			err      error
+		)
+		// The first iteration tries to use the informer's state.
+		if attempts == 0 {
+			existing, err = c.knCertificateLister.Certificates(desired.Namespace).Get(desired.Name)
+			existing = existing.DeepCopy()
+		} else {
```

Comment:
I want to a float an idea with an eye towards sharing this logic at some later point.

Given that we're already changing the signature (thanks, this has been on my TODO list 😅), what if we pass in the original informer copy instead of refetching it since it's right there?

Yes, this would mean slightly more conflicts (unclear how many), but I suspect that it would also greatly simplify this logic which we are copying all over.

Reply:
I'll have a look at it!

Reply:
Done the changes @mattmoor. It should not be possible to generalize this using the DynamicClient given a target GVR like you shoed in the PsReconciler. Was that the intent of the change?